### ISSUE TITLE
[WebGPU] Get incorrect output data of the Conv2d layer(name: squeezenet0_conv3_fwd) in SqueezeNet1.1 NCHW model

### ISSUE THREAD

**Issue by BruceDai** [tensorflow/tfjs#6284]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow.js): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow.js installed from (npm or script link): build tfjs-backend-webgpu from source
- TensorFlow.js version (use command below): 3.14.0
- Browser version: 101.0.4951.15 (Official Build) dev

**Describe the current behavior**
Running Image Classification with [SqueezeNet1.1 ONNX model](https://github.com/onnx/models/tree/master/vision/classification/squeezenet), we got different class label from the one by running WebGL backend.
We compared this output data by WebGPU backend with the output data by WebGL backend, we observed that there's the incorrect output data of the Conv2d layer(name: squeezenet0_conv3_fwd) by WebGPU backend. 

**Describe the expected behavior**
Running Image Classification with [SqueezeNet1.1 ONNX model](https://github.com/onnx/models/tree/master/vision/classification/squeezenet), should get same class label as the one by running WebGL backend.

![image](https://user-images.githubusercontent.com/1531742/165532429-0294967f-f793-421f-9741-83952caa58a8.png)

![image](https://user-images.githubusercontent.com/1531742/160985057-67900e78-e880-45ba-93c7-9f3e5df79ba8.png)

@qjia7 Please help look at this issue, thanks.

**Comment by BruceDai** [comment]
I ran SqueezeNet sample with the latest WebGPU backed @ [05a114e](https://github.com/tensorflow/tfjs/commit/05a114e7d2a84f36af29e667d3397e5b5e2a57b3), the issue still reproduced.

Comparing WebGL backend, I checked the output data of the Conv2d layer(name: squeezenet0_conv3_fwd) with/without Relu activation, I observed that those output data are similar without Relu activation while much different with Relu activation.
There would be an issue of `Conv2DMMVec4Program` since I could get correct result by `Conv2DMMProgram` .  

@qjia7 PTAL, thanks.

https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgpu/src/kernels/Conv2D_impl.ts#L297
```ts
    if (useVec4) {
      program = new Conv2DMMVec4Program(
          convInfo, hasBias, activation, hasPreluActivationWeights);
    } else {
      program = new Conv2DMMProgram(
          convInfo, hasBias, activation, hasPreluActivationWeights);
    }
```
